File: Java/Eta/ValueAdd/src/main/java/com/refinitiv/eta/valueadd/reactor/WatchlistStreamInfo.java
/*|-----------------------------------------------------------------------------
 *|            This source code is provided under the Apache 2.0 license      --
 *|  and is provided AS IS with no warranty or guarantee of fit for purpose.  --
 *|                See the project's LICENSE.md for details.                  --
 *|           Copyright (C) 2019-2022 Refinitiv. All rights reserved.         --
 *|-----------------------------------------------------------------------------
 */

package com.refinitiv.eta.valueadd.reactor;

/** 
 * Information about the stream associated with a message.
 * Only used when a watchlist is enabled.
 */
public class WatchlistStreamInfo
{
    String _serviceName;
    Object _userSpecObject;
    
    /**
     * Name of service associated with the stream, if any.
     * 
     * @return the service name
     */
    public String serviceName()
    {
        return _serviceName;
    }
    
    /**
     * Name of service associated with the stream, if any.
     * 
     * @param service name
     */
    void serviceName(String serviceName)
    {
        _serviceName = serviceName;
    }

    /**
     * User-specified object given when the stream was opened.
     * 
     * @return the user-specified object
     */
    public Object userSpecObject()
    {
        return _userSpecObject;
    }
    
    /**
     * User-specified object given when the stream was opened.
     * 
     * @param user-specified object
     */
    void userSpecObject(Object userSpecObject)
    {
        _userSpecObject = userSpecObject;
    }
    
    /**
     * Clears this object for reuse.
     */
    void clear()
    {
        _serviceName = null;
        _userSpecObject = null;
    }
}

File: CSharp/Eta/Src/ValueAdd/Reactor/WatchlistStreamInfo.cs
/*|-----------------------------------------------------------------------------
 *|            This source code is provided under the Apache 2.0 license
 *|  and is provided AS IS with no warranty or guarantee of fit for purpose.
 *|                See the project's LICENSE.md for details.
 *|           Copyright (C) 2023 LSEG. All rights reserved.     
 *|-----------------------------------------------------------------------------
 */

namespace LSEG.Eta.ValueAdd.Reactor
{
    /// <summary>
    /// Information about the stream associated with a message.
    /// </summary>
    public sealed class WatchlistStreamInfo
    {
        /// <summary>
        /// Gets or sets name of service associated with the stream, if any.
        /// </summary>
        public String? ServiceName { get; set; }

        /// <summary>
        /// Gets or sets user-specified object given when the stream was opened.
        /// </summary>
        public object? UserSpec { get; set; }

        /// <summary>
        /// Clears to default values
        /// </summary>
        public void Clear()
        {
            ServiceName = null;
            UserSpec = null;
        }
    }
}
